packages/yarnpkg-core/sources/structUtils.ts

```diff
@@ -669,8 +669,8 @@ export function prettyIdent(configuration: Configuration, ident: Ident): string
 
 function prettyRangeNoColors(range: string): string {
   if (range.startsWith(VIRTUAL_PROTOCOL)) {
-    const nested = prettyRangeNoColors(range.substr(range.indexOf(`#`) + 1));
-    const abbrev = range.substr(VIRTUAL_PROTOCOL.length, VIRTUAL_ABBREVIATE);
+    const nested = prettyRangeNoColors(range.substring(range.indexOf(`#`) + 1));
+    const abbrev = range.substring(VIRTUAL_PROTOCOL.length, VIRTUAL_ABBREVIATE);
```

Comment:
```suggestion
    const abbrev = range.substring(VIRTUAL_PROTOCOL.length, VIRTUAL_PROTOCOL.length + VIRTUAL_ABBREVIATE);
```

Reply:
This seems like a bugfix that should have a test associated with it.